src/Middleware/Diagnostics/src/ExceptionHandler/ExceptionHandlerMiddleware.cs

```diff
@@ -168,7 +176,10 @@ private static void ClearHttpContext(HttpContext context)
             // the endpoint and route values to ensure things are re-calculated.
             context.SetEndpoint(endpoint: null);
             var routeValuesFeature = context.Features.Get<IRouteValuesFeature>();
-            routeValuesFeature?.RouteValues?.Clear();
+            if (routeValuesFeature != null)
+            {
+                routeValuesFeature.RouteValues = null!;
+            }
```

Comment:
I think this should continue to clear. Otherwise, I'm pretty sure a new RouteValuesDictionary will be allocated per request.

Reply:
What's the difference between making our own copy vs capturing the original and nulling it out? Don't you end up with the same number of allocations?

Reply:
Ohh, this is happening in the exception filter and only if there is an exception.

In that case it doesn't matter.